### ISSUE TITLE
'unresolved external symbol' error. VS2015.

### ISSUE THREAD

**Issue by mloskot** [nanodbc/nanodbc#103]
_From @itisscan on March 24, 2016 7:47_

I can't build one project under the Visual Studio 2015 (Version 14.0.23107.0 D14REL) due to 'nanodbc' 2.12.4 version. 

I get  'unresolved external symbol' like this - 

`Error   LNK2001   unresolved external symbol "__declspec(dllimport) public: static class std::locale::id std::codecvt<char16_t,char,struct _Mbstatet>::id" (__imp_?id@?$codecvt@_SDU_Mbstatet@@@std@@2V0locale@2@A)   33_Urho2DSpriterAnimation   Github\Urho3D-master\build_1\Source\Samples\33_Urho2DSpriterAnimation\Urho3D_d.lib(nanodbc.obj)   1`

I have found out that is bug in vs2015. 

I see that in 'nanodbc' code is some workaround, but it seems do not work in my case. 

```
// Workaround for confirmed bug in VS2015.
// See: https://social.msdn.microsoft.com/Forums/en-US/8f40dcd8-c67f-4eba-9134 a19b9178e481/vs-2015-rc-linker-stdcodecvt-error

auto p = reinterpret_cast<wide_char_t const*>(in.data());
out = std::wstring_convert<NANODBC_CODECVT_TYPE<wide_char_t>, wide_char_t>().to_bytes(p, p + in.size());
```

UPDATE: 

I have tried to build nanodbc (static and dynamic)  under vs2015 as a standalone using its own build system. The result is the same.

If I build nanodbc as static, I get the 'unresolved external symbol' error - 
`Error   LNK2001   unresolved external symbol "__declspec(dllimport) public: static class std::locale::id std::codecvt<char16_t,char,struct _Mbstatet>::id" (__imp_?id@?$codecvt@_SDU_Mbstatet@@@std@@2V0locale@2@A)   example_northwind   D:\nanodbc-master\build\examples\nanodbc.lib(nanodbc.obj)   1`

If I build nanodbc as dynamic, I get the 'unresolved external symbol' error - 
`nanodbc.obj : error LNK2001: unresolved external symbol "__declspec(dllimport) public: static class std::locale::id std::codecvt<char16_t,char,struct _Mbstatet>::id" (__imp_?id@?$codecvt@_SDU_Mbstatet@@@std@@2V0locale@2@A)`

 In between times, I changed cmakelists.txt line 206 - 
`install(TARGETS nanodbc LIBRARY DESTINATION lib)` 
to
`install (TARGETS nanodbc
         ARCHIVE DESTINATION lib
         LIBRARY DESTINATION lib
         RUNTIME DESTINATION bin)`

in order to generate project as dynamic. (I use cmake 3.5.0)

_Copied from original issue: lexicalunit/nanodbc#121_

**Comment by mloskot** [comment]
~~This issue can be closed as it has already been fixed in the master by PR #96~~

I can confirm the issue with VS2015 Update 2 too.

**Comment by mloskot** [comment]
@itisscan Please, try this patch #125 and see if it fixes the problem for you.
